lib/nexmo_developer/app/presenters/page_title.rb

```diff
@@ -1,19 +1,59 @@
 class PageTitle
+  WORD_FILE = "#{Rails.configuration.docs_base_path}/config/normalized_words.yml".freeze
+
   def initialize(product, document_title)
     @product        = product
     @document_title = document_title
+
+    after_initialize!
+  end
+
+  def after_initialize!
+    raise 'You must provide a config/normalized_words.yml file in your documentation path.' unless File.exist?(WORD_FILE)
   end
 
   def title
     if @product && @document_title
-      "#{@product.titleize} > #{@document_title} | #{default_title}"
+      "#{normalized_product_title(@product)} > #{@document_title} | #{default_title}"
     elsif @document_title
       "#{@document_title} | #{default_title}"
     else
       default_title
     end
   end
 
+  def normalized_product_title(product)
+    if product.include?('/')
+      product_name = product.split('/')[0]
+      product_category = product.split('/')[1]
+      product_name_normalized = product_name.split(' ').map do |phrase|
+        phrase.split('-').map do |word|
+          normalized_word(word)
+        end
+      end
+      product_category_normalized = product_category.split(' ').map do |phrase|
+        phrase.split('-').map do |word|
+          normalized_word(word)
+        end
+      end
+      product = "#{product_name_normalized.join(' ')}/#{product_category_normalized.join(' ')}"
+    else
+      product = product.titleize
+    end
+    product
+  end
+
+  def normalized_word(word)
+    normalized_words = YAML.safe_load(File.open(WORD_FILE))
```

Comment:
This can be memoized by making it an instance variable meaning that it's loaded once per boot

```
@normalized_words ||= YAML.safe_load(File.open(WORD_FILE))
```